src/parser/get_executable_path.c

```diff
@@ -0,0 +1,57 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <libft.h>
+#include <parser/get_executable_path.h>
+
+static void	append_command_to_path(char	*buffer, const char *command)
+{
+	const int	cmd_len = ft_strlen(command) + TERMINATOR;
+
+	*buffer = FORWARD_SLASH;
+	buffer += 1;
+	ft_strlcpy(buffer, command, cmd_len);
+}
+
+static int	get_sub_path_len(const char *path)
+{
+	int	i;
+
+	i = 0;
+	while (path && path[i] != '\0' && path[i] != COLON)
+		i++;
+	return (i);
+}
+
+static void	get_single_path(const char *all_paths, char *buffer, int size)
```

Comment:
I think the name could be something like `copy_executable_path_to buffer`

Reply:
mmmm don't fully agree with this as the path is not executable, that is what we are checking after we've gotten the single_path, I was struggling to give this a proper name, maybe `copy_single_path_to_buffer` ?